android/src/main/java/com/appgoalz/rnjwplayer/RNJWPlayerView.java

```diff
@@ -290,62 +297,121 @@ public void setupPlayerView() {
-                    getReactContext().getJSModule(RCTEventEmitter.class).receiveEvent(
-                            getId(),
-                            "topFullScreenExit",
-                            eventExitFullscreen);
-                }
+    private class AppViewFullscreenHandler implements FullscreenHandler {
+        private ViewGroup mPlayerContainer;
+        private final RNJWPlayer mPlayer;
+        private View mDecorView;
 
-                @Override
-                public void onResume() {
+        public AppViewFullscreenHandler(RNJWPlayer player) {
+            mPlayerContainer = (ViewGroup) player.getParent();
+            mPlayer = player;
+        }
 
-                }
+        @Override
+        public void onFullscreenRequested() {
+            mDecorView = mActivity.getWindow().getDecorView();
 
-                @Override
-                public void onPause() {
+            // Hide system ui
+            mDecorView.setSystemUiVisibility(
+                View.SYSTEM_UI_FLAG_HIDE_NAVIGATION // hides bottom bar
+                | View.SYSTEM_UI_FLAG_FULLSCREEN // hides top bar
+                | View.SYSTEM_UI_FLAG_IMMERSIVE_STICKY // prevents navigation bar from overriding
+                // exit-full-screen button. Swipe from side to access nav bar.
+            );
 
-                }
+            // Enter landscape mode for fullscreen videos
+            mActivity.setRequestedOrientation(ActivityInfo.SCREEN_ORIENTATION_LANDSCAPE);
 
-                @Override
-                public void onDestroy() {
+            // Destroy the player's rendering surface, we need to do this to prevent Android's
+            // MediaDecoders from crashing.
+            mPlayer.destroySurface();
 
-                }
+            mPlayerContainer = (ViewGroup) mPlayer.getParent();
 
-                @Override
-                public void onAllowRotationChanged(boolean b) {
-                    Log.e(TAG, "onAllowRotationChanged: "+b );
-                }
+            // Remove the JWPlayerView from the list item.
+            if (mPlayerContainer != null) {
+                mPlayerContainer.removeView(mPlayer);
+            }
+
+            // Initialize a new rendering surface.
+            mPlayer.initializeSurface();
 
+            // Add the JWPlayerView to the RootView as soon as the UI thread is ready.
+            mRootView.post(new Runnable() {
                 @Override
-                public void updateLayoutParams(ViewGroup.LayoutParams layoutParams) {
-//        View.setSystemUiVisibility(int).
-//        Log.e(TAG, "updateLayoutParams: "+layoutParams );
+                public void run() {
+                    mRootView.addView(mPlayer, new ViewGroup.LayoutParams(
+                        ViewGroup.LayoutParams.MATCH_PARENT,
+                        ViewGroup.LayoutParams.MATCH_PARENT
+                    ));
+                    mFullscreenPlayer = mPlayer;
                 }
+            });
```

Comment:
```suggestion
            });
            
            WritableMap eventEnterFullscreen = Arguments.createMap();
            eventEnterFullscreen.putString("message", "onFullscreen");
            getReactContext().getJSModule(RCTEventEmitter.class).receiveEvent(
                    getId(),
                    "topFullScreen",
                    eventEnterFullscreen);

```

here too, add the onFullScreen props bridging